integration-tests/src/test/java/org/apache/druid/tests/coordinator/duty/ITConcurrentStreamAppendReplaceTest.java

```diff
@@ -0,0 +1,471 @@
+    final EventSerializer serializer = jsonMapper.readValue(
+        getResourceAsStream(Specs.SERIALIZER_PATH),
+        EventSerializer.class
+    );
+    streamGenerator = new WikipediaStreamEventStreamGenerator(serializer, 6, 100);
+  }
+
+  @Override
+  public String getTestNamePrefix()
+  {
+    return "autocompact_lock_contention";
+  }
+
+  @Test(dataProvider = "getParameters")
+  public void testConcurrentStreamAppendReplace(boolean transactionEnabled) throws Exception
+  {
+    if (shouldSkipTest(transactionEnabled)) {
+      return;
+    }
+
+    try (
+        final Closeable closer = createResourceCloser(generatedTestConfig);
+        final StreamEventWriter streamEventWriter = createStreamEventWriter(config, transactionEnabled)
+    ) {
+      // Start supervisor
+      final String taskSpec
+          = generatedTestConfig.getStreamIngestionPropsTransform()
+                               .apply(getResourceAsString(SUPERVISOR_WITH_CONCURRENT_LOCKS_SPEC_TEMPLATE_PATH));
+      generatedTestConfig.setSupervisorId(indexer.submitSupervisor(taskSpec));
+      LOG.info("supervisorSpec: [%s]", taskSpec);
+
+      // Generate data for minutes 1, 2 and 3
+      final Interval minute1 = Intervals.of("2000-01-01T01:01:00Z/2000-01-01T01:02:00Z");
+      long rowsForMinute1 = generateData(minute1, streamEventWriter);
+
+      final Interval minute2 = Intervals.of("2000-01-01T01:02:00Z/2000-01-01T01:03:00Z");
+      long rowsForMinute2 = generateData(minute2, streamEventWriter);
+
+      final Interval minute3 = Intervals.of("2000-01-01T01:03:00Z/2000-01-01T01:04:00Z");
+      long rowsForMinute3 = generateData(minute3, streamEventWriter);
+
+      // Wait for data to be ingested for all the minutes
+      ensureRowCount(rowsForMinute1 + rowsForMinute2 + rowsForMinute3);
+
+      ensureSegmentsLoaded();
+
+      // 2 segments for each minute, total 6
+      ensureSegmentsCount(6);
+      printTaskStatuses();
+
+      // Trigger auto compaction
+      submitAndVerifyCompactionConfig(null);
+      compactionResource.forceTriggerAutoCompaction();
+
+      // Verify that all the segments are now compacted
+      ensureRowCount(rowsForMinute1 + rowsForMinute2 + rowsForMinute3);
+      ensureSegmentsLoaded();
+      ensureSegmentsCount(3);
+      verifyCompactedIntervals(minute1, minute2, minute3);
+      printTaskStatuses();
+
+      // Concurrent compaction with configured segment granularity
+      for (int i = 0; i < 5; i++) {
+        rowsForMinute1 += generateData(minute1, streamEventWriter);
+        rowsForMinute2 += generateData(minute2, streamEventWriter);
+        rowsForMinute3 += generateData(minute3, streamEventWriter);
+
+        //ensureNoLockContention();
+        compactionResource.forceTriggerAutoCompaction();
+        ensureRowCount(rowsForMinute1 + rowsForMinute2 + rowsForMinute3);
+        printTaskStatuses();
+        checkAndSetConcurrentLocks();
+      }
+
+      // Verify the state with minute granularity
```

Comment:
Should the MINUTE and ALL granularity verifications be separate tests of their own?